pkg/provider/operators.go

```diff
@@ -73,32 +82,61 @@ func createOperators(csvs []olmv1Alpha.ClusterServiceVersion, subscriptions []ol
 			op.Package = subscription.Spec.Package
 			op.Org = subscription.Spec.CatalogSource
 			op.Channel = subscription.Spec.Channel
+			atLeastOneSubscription = true
 			break
 		}
-
-		csvInstallPlans, err := getCsvInstallPlans(csv.Namespace, csv.Name, installPlans)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get installPlans for csv %s (ns %s), err: %v", csv.Name, csv.Namespace, err)
+		if !atLeastOneSubscription {
+			logrus.Warnf("Subscription not found for csv %s (ns %s) This Operator will not receive updates.", csv.Name, csv.Namespace)
 		}
 
-		for _, installPlan := range csvInstallPlans {
-			indexImage, catalogErr := getCatalogSourceImageIndexFromInstallPlan(installPlan)
+		atLeastOneInstallPlan := false
+		for _, installPlan := range allInstallPlans {
+			if installPlan.Namespace != csv.Namespace {
+				continue
+			}
+			alLeastOneCsv := false
+			for _, csvName := range installPlan.Spec.ClusterServiceVersionNames {
+				if csv.Name != csvName {
+					continue
+				}
+
+				if installPlan.Status.BundleLookups == nil {
+					logrus.Warnf("InstallPlan %s for csv %s (ns %s) does not have bundle lookups. It will be skipped.", installPlan.Name, csv.Name, csv.Namespace)
+					continue
+				}
+				alLeastOneCsv = true
+				break
+			}
+			if !alLeastOneCsv {
+				continue
+			}
+			indexImage, catalogErr := getCatalogSourceImageIndexFromInstallPlan(installPlan, allCatalogSources)
 			if catalogErr != nil {
-				return nil, fmt.Errorf("failed to get installPlan image index for csv %s (ns %s) installPlan %s, err: %v",
+				logrus.Tracef("failed to get installPlan image index for csv %s (ns %s) installPlan %s, err: %v",
 					csv.Name, csv.Namespace, installPlan.Name, catalogErr)
+				continue
 			}
 
 			op.InstallPlans = append(op.InstallPlans, CsvInstallPlan{
 				Name:        installPlan.Name,
 				BundleImage: installPlan.Status.BundleLookups[0].Path,
 				IndexImage:  indexImage,
 			})
+			atLeastOneInstallPlan = true
+			break
+		}
+		if !atLeastOneInstallPlan {
+			logrus.Warnf("InstallPlan with BundleLookups not found for csv %s (ns %s) not present. Catalog source not available", csv.Name, csv.Namespace)
+		}
+		if !(atLeastOneInstallPlan || catalogSourceNotRequired) {
+			continue
+		}
+		if !(csv.Status.Phase == olmv1Alpha.CSVPhaseSucceeded || !succeededRequired) {
```

Comment:
Shouldn't this check be done at the very beginning (line 65)?

Reply:
fixed